bigchaindb/common/utils.py

```diff
@@ -46,3 +49,17 @@ def deserialize(data):
             string.
     """
     return rapidjson.loads(data)
+
+
+def validate_asset_data_keys(tx_body):
+    backend = bigchaindb.config['database']['backend']
+
+    if backend == 'mongodb':
+        data = tx_body['asset'].get('data', {})
+        keys = data.keys() if data else []
+        for key in keys:
```

Comment:
This only checks the top level keys. This will not fail for something like:
```json
{
  "data": {
    "msg": {
      "$nested": "hello"
    }
  }
}
```